src/os/linux/local/mode/ntp.pm

```diff
@@ -209,39 +232,166 @@ sub get_ntp_modes {
+    # timedatectl show-timesync to retrieve 'SystemNTPServers', 'FallbackNTPServers', 'ServerAddress' and 'ServerName'
+    my ($stdout_status) = $options{custom}->execute_command(    command => 'timedatectl',
+                                                                command_options => 'status 2>&1',
+                                                                no_quit => 1, );
+    my ($stdout_timesync) = $options{custom}->execute_command(  command => 'timedatectl',
+                                                                command_options => 'timesync-status 2>&1',
+                                                                no_quit => 1, );
+    my ($stdout_show) = $options{custom}->execute_command(      command => 'timedatectl',
+                                                                command_options => 'show-timesync 2>&1',
+                                                                no_quit => 1, );
+
+    my %values = ( ( map { /^\s*(.+?): (.+)$/ } split /\n/, $stdout_timesync ),
+                   ( map { /^(.+?)=(.+)$/ } split /\n/, $stdout_show ),
+                   ( map { /^\s*(.+?): (.+)$/ } split /\n/, $stdout_status ) );
+
+    return "timedatectl not available" unless $values{'NTP service'};
+
+    $values{$_}//='' foreach ('SystemNTPServers', 'FallbackNTPServers', 'ServerAddress', 'ServerName', 'Offset', 'Stratum', 'Packet count', 'System clock synchronized');
+
+    my $active_is_fallback = 0;
+
+    # Primary and fallback servers are initialized, the active server is excluded as it will be initialized later with additional information
+    # A server has either the type 'primary' or 'fallback'
+    # A 'primary' server can have the states 'available', 'synchronized', 'syncing' or 'inactive'
+    # A 'fallback' server can have the states 'unused', synchronized', 'syncing' or 'inactive'
+    foreach (split /\s/, $values{SystemNTPServers}) {
+        next if $values{ServerAddress} && ($_ eq $values{ServerAddress} || $_ eq $values{ServerName});
+
+        $self->{peers}->{$_} = { display => $_, rawstate => 'available', stratum => '', rawtype => 'primary', reach => '', offset => '' };
+        $self->{global}->{peers}++
+    }
+    foreach (split /\s/, $values{FallbackNTPServers}) {
+        if ($values{ServerAddress} && ($_ eq $values{ServerAddress} || $_ eq $values{ServerName})) {
+            $active_is_fallback = 1;
+            next
+        }
+        $self->{peers}->{$_} = { display => $_, rawstate => 'unused', stratum => '', rawtype => 'fallback', reach => '', offset => '' };
+        $self->{global}->{peers}++
+    }
+
+    if ($values{ServerAddress} ne '') {
+        # If there is an active server it is initialized here with all information
+        $values{ServerName} = $values{ServerAddress} if $values{ServerName} eq '';
+        $values{'Offset'} = $1 * $unit_map_timedatectl{$2} if $values{'Offset'} =~ /^(.*?)([a-z]+)$/ && $unit_map_timedatectl{$2};
```

Comment:
```suggestion
        $values{'Offset'} = $1 * $unit_map_timedatectl{$2} if ( $values{'Offset'} =~ /^(.*?)([a-z]+)$/ && defined($unit_map_timedatectl{$2}) );
```

Reply:
I know that it will result in the same behaviour given none of the hash's keys is equal to 0 but I think it is a better practice to test it this way since a value of 0 would give false in this condition